omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -546,26 +546,38 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
             });
 
             // Browser back button support, for preserve data
-            if (lastSearchParams !== null) {
-                $j('#inputNode').val(lastSearchParams.lastSearchText);
-                if (lastSearchParams.includeVoided == 1) {
+            var lastSearchText = decodeURIComponent((new RegExp('[?|&]' + 'lastSearchText' 
+            		+ '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20')) || null;
+            var includeVoided = decodeURIComponent((new RegExp('[?|&]' + 'includeVoided' 
+            		+ '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20')) || null;
+            var includeVerbose = decodeURIComponent((new RegExp('[?|&]' + 'includeVerbose' 
+            		+ '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20')) || null;
+            
+            //if we have lastSearchText then set input text, move cursor to the end and trigger a keyup event to load the search
+            if (lastSearchText !== null) {
+            	
+                $j('#inputNode').val(lastSearchText);
+                if (includeVoided == 1) {
                     $j('#includeVoided').attr('checked','checked');
                 }
-                if (lastSearchParams.includeVerbose == 1) {
+                if (includeVerbose == 1) {
                     $j('#includeVerbose').attr('checked','checked');
                 }
-                var keyEvent = jQuery.Event("keyup");
-
-                var codex = 13;
-
-                // Patient search doesn't support for Enter Key. So invoke keyup
-                // event with last character of searched text.
-                if (el.attr('id') === "findPatients") {
-                    var text = $j('#inputNode').val();
-                    codex = $j('#inputNode').val().charCodeAt($j('#inputNode').val().length - 1);
-                }
-
-                keyEvent.keyCode = codex;
+                
+                //setting the cursor to the end of search text
+    		//setSelectionRange is a method of the DOM element not the jQuery object so get the DOM element
+		input.focus();
+        	var inputNodeDOM = $j('#inputNode').get(0);
+        	if(inputNodeDOM){
+        		var lastSearchTextLength = lastSearchText.length;
+        		if (inputNodeDOM.setSelectionRange) {
+        			inputNodeDOM.setSelectionRange(lastSearchTextLength,lastSearchTextLength);
+        		}
+        	}
+       
+                var keyEvent = jQuery.Event("keyup");    
+                //jQuery normalizes event.which fixing some issues with not being able to trigger enter key 
+                keyEvent.which = 13;
                 $j("#inputNode").trigger(keyEvent);
             }
 
```

Comment:
Seems to me like we are doing more than what needs to be done to achieve this, the widgets already have a way to support triggering a search on page load given an initial search phrase, it configured as searchPhrase in the widget options, you should just read the last search text from the browser history and set it as the value of the searchPhrase widget option and things will happen magically